core/src/main/java/com/adobe/aem/commons/assetshare/content/renditions/impl/dispatchers/StaticRenditionDispatcherImpl.java

```diff
@@ -110,12 +116,20 @@ public boolean isHidden() {
     }
 
     @Override
-    public void dispatch(SlingHttpServletRequest request, SlingHttpServletResponse response) throws IOException {
+    public void dispatch(SlingHttpServletRequest request, SlingHttpServletResponse response) throws IOException, ServletException {
         final Asset asset = DamUtil.resolveToAsset(request.getResource());
         final AssetRenditionParameters parameters = new AssetRenditionParameters(request);
 
         final Rendition rendition = asset.getRendition(new PatternRenditionPicker(mappings.get(parameters.getRenditionName())));
 
+        if (requireAem.isRunningInAdobeCloud()) {
+            dispatchFromAemCloud(rendition, parameters, response);
+        } else {
+            dispatchFromAem(rendition, parameters, response);
+        }
+    }
+
```

Comment:
@davidjgonzalez Redirecting to the rendition path (taking into account the potential context path) will allow the same code to work on both AEM and AEM Cloud.

Reply:
@bobvanmanen who can I set if it will be downloaded as an attachment or not using this method? Can I set them on the response before sending it through?

Reply:
It's always attachment, which is intentional and works with the `<img>` tag but ensures the image is downloaded when requested directly. Is there a use case where this would not work?

Reply:
well - asset share commons is for any asset type; pdf for example is a great one you often want to open in the browsers viewer. There are many other asset types that benefit from this as well.

Even Images can be handy to open in the browser for a quick full preview. tbh I’m surprised sling would be so opinionated about this.